owncloudData/src/test/java/com/owncloud/android/data/transfers/implementation/OCLocalTransferDataSourceTest.kt

```diff
@@ -0,0 +1,505 @@
+        } throws Exception()
+
+        ocLocalTransferDataSource.updateTransferStorageDirectoryInLocalPath(id, oldDirectory, newDirectory)
+
+    }
+    @Test
+    fun `deleteTransferById returns a unit`() {
+        every {
+            transferDao.deleteTransferWithId(id)
+        } returns Unit
+
+        ocLocalTransferDataSource.deleteTransferById(id)
+
+        verify(exactly = 1) {
+            transferDao.deleteTransferWithId(id)
+        }
+    }
+
+    @Test(expected = Exception::class)
+    fun `deleteTransferById returns an exception when dao receive an exception`() {
+        every {
+            transferDao.deleteTransferWithId(id)
+        } throws Exception()
+
+        ocLocalTransferDataSource.deleteTransferById(id)
+
+    }
+    @Test
+    fun `deleteAllTransfersFromAccount returns a unit`() {
+        every {
+            transferDao.deleteTransfersWithAccountName(OC_ACCOUNT_NAME)
+        } returns Unit
+
+        ocLocalTransferDataSource.deleteAllTransfersFromAccount(OC_ACCOUNT_NAME)
+
+        verify(exactly = 1) {
+            transferDao.deleteTransfersWithAccountName(OC_ACCOUNT_NAME)
+        }
+    }
+
+    @Test(expected = Exception::class)
+    fun `deleteAllTransfersFromAccount returns an exception when dao receive an exception`() {
+        every {
+            transferDao.deleteTransfersWithAccountName(OC_ACCOUNT_NAME)
+        } throws Exception()
+
+        ocLocalTransferDataSource.deleteAllTransfersFromAccount(OC_ACCOUNT_NAME)
+
+    }
+    @Test
+    fun `getTransferById returns a OCTransfer`() {
+        every {
+            transferDao.getTransferWithId(any())
+        } returns ocTransfer.toEntity()
+
+        val actualResult = ocLocalTransferDataSource.getTransferById(id)
+
+        assertEquals(ocTransfer, actualResult)
+
+        verify(exactly = 1) {
+            transferDao.getTransferWithId(id)
+        }
+    }
+
+    @Test(expected = Exception::class)
+    fun `getTransferById returns an exception when dao receive an exception`() {
+        every {
+            transferDao.getTransferWithId(any())
+        } throws Exception()
+
+        ocLocalTransferDataSource.getTransferById(id)
+
+    }
+
+    @Test
+    fun `getAllTransfers returns a list of OCTransfer`() {
+
+        every {
+            transferDao.getAllTransfers()
+        } returns listOf(transferEntity)
+
+        val actualResult = ocLocalTransferDataSource.getAllTransfers()
+
+        assertEquals(listOf(transferEntity.toModel()), actualResult)
+
+        verify(exactly = 1) {
+            transferDao.getAllTransfers()
+        }
+    }
+
+    @Test(expected = Exception::class)
+    fun `getAllTransfers returns an exception when dao receive an exception`() {
+
+        every {
+            transferDao.getAllTransfers()
+        } throws Exception()
+
+        ocLocalTransferDataSource.getAllTransfers()
+
+    }
+
+    @Test
+    fun `getAllTransfersAsStream returns a flow of list of OCTransfer`() = runBlocking {
```

Comment:
Here the key is testing that the transfers are retrieved in the specified order following their status, so:
1) In progress
2) Queued
3) Failed
4) Succeded

Reply:
Done